### ISSUE TITLE
Delay in mergeStyle taking effect

### ISSUE THREAD

**Issue by jwnimmer-tri** [Reviewable/Reviewable#736]
In https://github.com/RobotLocomotion/drake/ we use the `mergeStyle` option in our completion condition, keyed on a label:
```js
var properlyCurated = review.labels.indexOf('status: commits are properly curated') >= 0;
return {
  ...
  mergeStyle: properlyCurated ? 'merge' : 'squash',
  ...
}
```

We also use the `:lgtm:` emoji for approvals, not the github review approvals.

Our GH repository is set to allow either merge commits or squash merges (no rebases).

The full text of the condition is shown at
https://github.com/RobotLocomotion/drake-ci/wiki/reviewable.io-review-completion-condition

I'm seeing a problem when the final assigned reviewer clicks the `:lgtm:` and then submits their comment.

The review button at the top right of the page goes green to permit merging, but for about two seconds it shows "MERGE & RETAIN BRANCH" before switching to "SQUASH & RETAIN BRANCH".  If I quickly open the drop-down during that time, I see both radio buttons enabled for a moment, until after a couple seconds the merge radio option disables and the choice switches to squash option.

In short, there appears to be some kind of race condition when reflecting the mergeStyle into the merge widget.

Note that the `mergeStyle` value is not fluttering during the review -- it's set up front based on a label and the label almost never changes during the course of the review -- so my condition's returned value for this key should be consistent, even though `completed` returned bool changes.

Some of our most frequent reviewers are obviously the fastest at clicking through "LGTM" then "Merge" so end up merging using the wrong style.

**Comment by pkaminski** [comment]
@dhbradshaw, want to take a look at this?

**Comment by dhbradshaw** [comment]
Interesting.  I'll see what I can figure out.

**Comment by dhbradshaw** [comment]
And, this should now be resolved!  Test it out @jwnimmer-tri and let us know if it solves the issue you were seeing!
